cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/service/StorageProviderNamespaceAdmin.java

```diff
@@ -57,17 +59,31 @@ protected StorageProviderNamespaceAdmin(CConfiguration cConf, NamespacedLocation
    */
   protected void create(Id.Namespace namespaceId) throws IOException, ExploreException, SQLException {
     Location namespaceHome = namespacedLocationFactory.get(namespaceId);
-    if (namespaceHome.exists()) {
-      LOG.warn("Home directory '{}' for namespace '{}' already exists. Deleting it.",
-               namespaceHome, namespaceId);
-      if (!namespaceHome.delete(true)) {
-        throw new IOException(String.format("Error while deleting home directory '%s' for namespace '%s'",
+    // TODO (CDAP-6155): This is not clean. Once we do hbase mapping the DatasetFramework will pass down the
+    // NamespaceMeta to this create and we should look for the custom location in the config rather than
+    // checking like this.
+    if (namespaceHome.toString().endsWith(Joiner.on(File.separator).join(cConf.get(Constants.Namespace.NAMESPACES_DIR),
```

Comment:
can't we instead pass the ``NamespaceConfig`` here somehow? String-matching doesn't seem fool-proof?